### PR TITLE
[cdc-base] Release the debezium reader thread resources after reading finished

### PR DESCRIPTION
**Pull request by ruanhang1993**
Apply changes in #1358 to cdc-base.

### CODE REVIEW

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java [hunk 1]
```diff
@@ -134,7 +137,21 @@ private void checkReadException() {
...
+                executorService.shutdown();
+                if (executorService.awaitTermination(
+                        READER_CLOSE_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                    LOG.warn(
+                            "Failed to close the binlog split reader in {} seconds.",
+                            READER_CLOSE_TIMEOUT_SECONDS);
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Close binlog reader error", e);
```
Comment (Jiabao-Sun) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java:152]:
```suggestion
            LOG.error("Close stream fetcher error", e);
```

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java [hunk 2]
```diff
@@ -134,7 +137,21 @@ private void checkReadException() {
...
-    public void close() {}
+    public void close() {
+        try {
+            if (executorService != null) {
+                executorService.shutdown();
+                if (executorService.awaitTermination(
+                        READER_CLOSE_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                    LOG.warn(
+                            "Failed to close the binlog split reader in {} seconds.",
+                            READER_CLOSE_TIMEOUT_SECONDS);
```
Comment (Jiabao-Sun) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java:148]:
```suggestion
                    LOG.warn(
                            "Failed to close the stream fetcher in {} seconds.",
                            READER_CLOSE_TIMEOUT_SECONDS);
```

**Review by Jiabao-Sun — COMMENTED** [review]
Thanks @ruanhang1993 for the contribution.
I left few comments.

**Comment by ruanhang1993** [comment]
Thanks for your review @Jiabao-Sun . Fixed.

**Review by Jiabao-Sun — APPROVED** [review]
Thanks @ruanhang1993. 
LGTM. Will merge when CI passed.

**Pull request merged**